zombies.js

```diff
@@ -80,6 +98,121 @@
  */
 
 
+class Player {
+  constructor(name, health, strength, speed){
+    this.name = name;
+    this.health = health;
+    this.strength = strength;
+    this.speed = speed;
+    this._pack = [];
+    this._maxHealth = health;
+    this.isAlive = true;
+    this.equipped = false;
+
+  }
+  getPack (){
+    return this._pack;
+  }
+
+  getMaxHealth(){
+    return this._maxHealth;
+  }
+
+  checkPack(){
+    console.log(this._pack);
+  }
+
+
+  takeItem(item){
+    if(this.getPack().length <3 ){
+      this._pack.push(item);
+      console.log("Player " + this.name + "\'s your item, " + this.item + ", has been added to your pack")
+      return true;
+    }else{
+      console.log("Sorry, your pack is full");
+      return false;
+    }
+  }
+
+
+
+  discardItem(item){
+    let itemIndex = this.getPack().indexOf(item); // index of item to take off from pack
+    if(itemIndex === -1){
+      console.log("Player " + this.name + "\'s item," + this.item + " ,is not found in pack, no item discarded");
+      return false;
+    }else{
+      var removeItem = this.getPack().splice(itemIndex, 1); // item taken from bag, then remove 1
+      console.log("Player " + this.name + "\'s item, " + this.item + " ,has been discarded");
+      return true;
+    }
+  }
+
+
+
+  equip(weapon){
+    let weaponIndex = this.getPack().indexOf(weapon);
+    if (weapon instanceof Weapon){ // if weapon is within the 'WEAPON'
+      if (weaponIndex  === -1){ // if it is currently not in pack
+//        if (this.equipped === false) // if not equipped
```

Comment:
You do need to test this.equipped before moving the currently equipped weapon into the pack